src/Team-Services-Bot.Api/Dialogs/ConnectDialog.cs

```diff
@@ -65,8 +64,15 @@ public Task StartAsync(IDialogContext context)
 
         private static string GeneratePin()
         {
-            var generator = new Random();
-            return generator.Next(Min, Max).ToString("0000", CultureInfo.InvariantCulture);
+            using (var generator = new RNGCryptoServiceProvider())
+            {
+                var data = new byte[4];
+
+                generator.GetBytes(data);
+                var value = BitConverter.ToUInt16(data, 0);
```

Comment:
The generator requires at least four bytes. I am only looking for a random number, to start the dialog again, after the user moved to the login webpage.

But I will update it with your suggestion.

Regarding the unit test. As said by Vladimir. It is a private method and needs to be tested within a public method. Which I want pickup after we got our unit test approach.